Add MSI installer

gentoo90 opened the pull request:
* adds GitTfs to `%PATH%` on install and removes on uninstall
* supports upgrade
* uses [WiX#](https://wixsharp.codeplex.com/) ([WiX](http://wixtoolset.org/) should be installed to build this)

Just build `GitTfs.Setup` project and MSI file will be in its root folder

pmiossec commented:
We need to discuss about that with @spraints because our chocolatey package already provide these features and is easier to build. 

It will need to do the job twice :-(

spraints commented:
an MSI for git tfs seems fine to me. What does the dependency story look like? I think the only chocolatey dep is Git.

spraints commented on GitTfs.Setup/Program.cs:
```diff
@@ -0,0 +1,74 @@
+            var ProjectName = "GitTfs";
+            var GUID = new Guid("98823CC2-0C2E-4CF7-B5ED-EA2DD26559BF");
+            var Conf = "Release";
+#endif
```
Is there a way to do this without a `#ifdef`? I'd rather avoid preprocessor stuff.

spraints replied:
If this is generated code, please feel free to ignore this comment.

gentoo90 replied:
I'm not aware of any other way to distinguish Debug and Release. Anyway why are you against `#if DEBUG`? Is preprocessor in C# broken?

spraints replied:
I avoid preprocessor conditionals because they make it so that different code is compiled in debug and release. It makes it easier to accidentally add a bug that affects release without noticing the problem in debug.

gentoo90 commented:
Rebased and updated. I believe
```C#
#if DEBUG
    var IsDebug = true;
#else
    var IsDebug = false;
#endif
    if (IsDebug) ...
```
is the minimum possible usage of preprocessor.

spraints commented:
:sunglasses: Looks good.

gentoo90 commented:
Appveyor doesn't want to add `msi` artifact :(
Although [it did for my repo](https://ci.appveyor.com/project/gentoo90/git-tfs/build/artifacts) and with the same commit.

pmiossec commented:
It seems to generate an exe ( GitTfs.Setup ->
C:\gittfs\GitTfs.Setup\bin\Release\GitTfs.Setup.exe) and you are looking
for a msi to push as artefact...

gentoo90 commented:
`C:\gittfs\GitTfs.Setup\bin\Release\GitTfs.Setup.exe` runs WiX which builds MSI.

spraints commented:
When I made v0.25.0 just now, my build machine said it couldn't find wix. Is there a way to install wix via paket?

pmiossec commented:
.\.paket\paket.exe add nuget wix

should work (and you will have to commit the 2 packet files updated)
